src/vaillant_netatmo_api/thermostat.py

```diff
@@ -572,13 +572,21 @@ def __init__(
         temperature: float | None = None,
         setpoint_temp: float | None = None,
         est_setpoint_temp: float | None = None,
+        gas_heating_usage: list[float] | None = None,
+        gas_water_usage: list[float] | None = None,
+        elec_heating_usage: list[float] | None = None,
+        elec_water_usage: list[float] | None = None,
         **kwargs,
     ) -> None:
         """Create new measured attribute."""
 
         self.temperature = temperature
         self.setpoint_temp = setpoint_temp
         self.est_setpoint_temp = est_setpoint_temp
+        self.gas_heating_usage = gas_heating_usage
+        self.gas_water_usage = gas_water_usage
+        self.elec_heating_usage = elec_heating_usage
+        self.elec_water_usage = elec_water_usage
```

Comment:
Lets discuss this 😄 

According to the HA [docs](https://developers.home-assistant.io/docs/api_lib_data_models), libraries should try to design their public API with as little changes as possible compared to the underlying API the library is a wrapper for.

With that in mind, I suggest we don't add these new data points to the `Measured` class. This object is returned when library gets data from `getthermostatsdata` API endpoint. AFAIK this endpoint doesn't return these newly added data points, so we'd be violating the HA practices as we'd have properties that don't exist in the underlying API call.

When exposing these new data points in HA, we can expose them as separate energy sensors, with their own lifecycle of fetching the data using the `getmeasure` API endpoint. That way library stays clean (ie. almost exact output as it would be calling the APIs directly through `curl`) and HA integration can decouple these new sensors from the already implemented `Climate` entity.

What do you think about this?

Reply:
I've added new class for EnergyUsage. Now it can be exposed as separate energy sensor with their own lifecycle of fetching the data. Please review my changes.